spec/unit/managed_files_database_spec.rb

```diff
@@ -0,0 +1,250 @@
+
+require_relative "spec_helper"
+
+describe ManagedFilesDatabase do
+  describe ManagedFilesDatabase::ChangedFile do
+    describe "config_file?" do
+      it "returns true for config files" do
+        file = ManagedFilesDatabase::ChangedFile.new(
+          "c",
+          name:    "/etc/foo",
+          status:  "changed",
+          changes: ["md5"]
+        )
+        expect(file.config_file?).to be(true)
+      end
+
+      it "returns false for non-config files" do
+        file = ManagedFilesDatabase::ChangedFile.new(
+          "",
+          name:    "/etc/foo",
+          status:  "changed",
+          changes: ["md5"]
+        )
+        file.type = ""
+        expect(file.config_file?).to be(false)
+      end
+    end
+  end
+
+  let(:system) { LocalSystem.new }
+  let(:changed_files_sh_result) {
+    File.read(File.join(Machinery::ROOT,
+      "spec/data/rpm_managed_files_database/changed_files_sh_result"))
+  }
+  subject { ManagedFilesDatabase.new(system) }
+
+  describe "#changed_files" do
+    before(:each) do
+      allow_any_instance_of(ManagedFilesDatabase).to receive(:check_requirements)
+      allow_any_instance_of(ManagedFilesDatabase).to receive(:managed_files_list).
+        and_return(changed_files_sh_result)
+      allow_any_instance_of(ManagedFilesDatabase).to receive(:package_for_file_path).
+        and_return(["zypper", "1.6.311"])
+
+      allow(system).to receive(:run_command).with("stat", any_args).and_return(
+        File.read(File.join(Machinery::ROOT, "spec/data/rpm_managed_files_database/stat_result"))
+      )
+      allow(system).to receive(:run_command).with("find", any_args).and_return(
+        "/link_target"
+      )
+    end
+
+    it "returns the list of files" do
+      expected = ManagedFilesDatabase::ChangedFile.new(
+        "",
+        name:            "/etc/iscsi/iscsid.conf",
+        status:          "changed",
+        changes:         ["size", "mode", "md5", "user", "group", "time"],
+        package_name:    "zypper",
+        package_version: "1.6.311",
+        user: "root",
+        group: "root",
+        mode: "6644",
+        type: "file"
+      )
+
+      expect(subject.changed_files.first).to eq(expected)
+      expect(subject.changed_files.length).to eq(7)
+    end
+  end
+
+  describe "parse_changes_line" do
+    capture_machinery_output
+
+    it "parses an md5 change" do
+      line = "..5......  c /etc/pulse/client.conf"
+      file, changes, flag = subject.parse_changes_line(line)
+      expect(file).to eq("/etc/pulse/client.conf")
+      expect(changes).to match_array(["md5"])
+      expect(flag).to eq("c")
+    end
+
+    it "parses all changes" do
+      line = "SM5DLUGTP  c /etc/pulse/client.conf"
+      file, changes, flag = subject.parse_changes_line(line)
+      expect(file).to eq("/etc/pulse/client.conf")
+      expect(changes).to match_array([
+        "size", "mode", "md5", "device_number", "link_path", "user", "group", "time", "capabilities"
```

Comment:
Use 2 spaces for indentation in an array, relative to the first position after the preceding left parenthesis.